### ISSUE TITLE
File extensions should not be changed on save

### ISSUE THREAD

**Issue by JavaNut13** [MacDownApp/macdown#600]
MacDown should not change the file extension of a file when it is saved. This is very frustrating when working with files that have an 'outer' extension as well as the normal `.md` extension.

To reproduce:

1. Open a file that has an extension that is not `.md` (i.e. `my_file.md.erb`)
2. Save the file
3. Observe the file extension changing

A possible solution is to provide an option to disable this, or to only change the file extension if there is no `.md` anywhere in the title, so that it will not affect `.erb` files.

**Comment by Jmuccigr** [comment]
I don't see this happening with v. 0.6 on Mavericks.

**Comment by uranusjr** [comment]
Me either, and I think this might have something to do with system settings. Do you show file extensions in Finder?

![screen shot 2016-04-07 at 23 56 31](https://cloud.githubusercontent.com/assets/605277/14357769/9c9434a8-fd1c-11e5-8177-2fbffa15abf7.png)

**Comment by JavaNut13** [comment]
Yes, I have filenames shown. I am on El Capitan with MacDown 0.6.

**Comment by massenz** [comment]
I have seen something similar when renaming a file (from .rst to .md).
In general, I would recommend leaving the file extensions alone when saving a file; in other words, if there is NO extension, then add `.md`, but if there is one already, do not do anything.

Also, it seems that during "Save As..." the extension cannot be edited - this also should be allowed, and the resulting editing filename should be left alone.

I am sorry I cannot remember exactly the sequence, but in the end, I had to close MacDown, edit the filename in Finder, and then re-open.

(BTW - this has nothing to do with Finder's "hide extension" thing - I use the shell (Terminal) to manipulate my files, so I can always see the 'raw' filenames).

**Comment by Jmuccigr** [comment]
Nevertheless the original report seems difficult to reproduce.

I also just did a save-as and changed the extension in the dialog box successfully. I didn't see the issue @massenz reports.

**Comment by Jmuccigr** [comment]
I've just upgraded to El Capitan. What I observe is that if I have a file open called "macdown.md.erb" and use the "save as" command, the dialog that opens drops the ".erb" extension, leaving just "md", but I can add it without a problem.

**Comment by franklinyu** [comment]
I failed to reproduce this issue as well. I created `my_file.md.erb` with some content, opened it with MacDown, added some lines, and saved it without issue. @JavaNut13 Does the bug still occurs recently?

**Comment by massenz** [comment]
On MacOS Sierra:

1. with an existing MD file (you can't really "Save As" but you can "Duplicate")
- open an existing file (`README.md`);
- File > Duplicate
- Filename is now editable (defaults to `README copy`) in the top bar;
- edit the name to be `README copy.foo`
- observe the name (in Terminal) to be named `README copy.foo.md`

I suspect this may a MacOS thing - in that it does not let you edit the file's extension, but only the name (depending on your settings, you may or may not observe it in Finder too -- on my MacPro, Finder shows `README copy.foo`; which may also be a bug, as just below I see `README.md` - but whatever: use Terminal, people)

2. with a new file:
- works as intended (saving it with a random extension leaves it alone)

3. with a non-MD file:
- working as intended (also Rename works just fine).

Generally speaking, I'd say that we can consider the issue closed.

**Comment by Jmuccigr** [comment]
Isn't the ability to save as instead of duplicate? I'm not on Sierra yet, but I can do both on ElCap.
